include/modem/sms.h

```diff
@@ -24,11 +24,82 @@ extern "C" {
 #include <zephyr/types.h>
 #include <sys/types.h>
 
+enum sms_type {
+	SMS_TYPE_DELIVER = 0,
+	SMS_TYPE_SUBMIT_REPORT
+};
+
+#define SMS_MAX_ADDRESS_LEN_OCTETS 10
+#define SMS_MAX_ADDRESS_LEN_CHARS (2 * SMS_MAX_ADDRESS_LEN_OCTETS)
+
+/* Forward declaration */
+struct sms_deliver_header;
+
 /** @brief SMS PDU data. */
 struct sms_data {
 	char *alpha;
 	uint16_t length;
 	char *pdu;
+	enum sms_type type;
+	struct sms_deliver_header *header;
```

Comment:
Right now this entire structure is organized for receiving SMS-DELIVER message which definitely is the most important message for SMS. However, There are status reports also which we could support and even if we wouldn't do that support instantly, this is not very extendable for that purpose.

Reply:
This is refactored based on a meeting